examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/main.c

```diff
@@ -356,58 +331,69 @@ void on_connected( xi_context_handle_t in_context_handle, void* data, xi_state_t
-                    void* user_data )
+void on_led_msg( xi_context_handle_t in_context_handle,
+                 xi_sub_call_type_t call_type,
+                 const xi_sub_call_params_t* const params,
+                 xi_state_t state,
+                 void* user_data )
 {
-    /* sanity checks */
-    assert( NULL != user_data );
-
-    /* user data is the mqtt_topic_configuration_t */
-    const mqtt_topic_configuration_t* topic_conf =
-        ( const mqtt_topic_configuration_t* )user_data;
-
     switch ( call_type )
     {
         case XI_SUB_CALL_SUBACK:
             if ( XI_MQTT_SUBACK_FAILED == params->suback.suback_status )
             {
-                printf( "Subscription failed for %s topic\r\n", params->suback.topic );
+                printf( "Subscription failed for LED topic\r\n" );
             }
             else
             {
-                printf( "Subscription successful for %s topic\r\n",
+                printf( "Subscription successful for LED topic\r\n",
                         params->suback.topic );
             }
             return;
         case XI_SUB_CALL_MESSAGE:
-            printf( "received message on %s topic\r\n", params->suback.topic );
-            /* if there is a handler call it with the message payload */
-            if ( NULL != topic_conf->on_msg_pull )
+            printf( "received message on LED topic\r\n" );
+            printf( "\r\n\t Message size: %d",
+                    params->message.temporary_payload_data_length );
+            printf( "\r\n\t Message payload: " );
+            for ( size_t i = 0; i < params->message.temporary_payload_data_length; i++ )
+            {
+                printf( "0x%02x ", params->message.temporary_payload_data[i] );
+            }
+
+            if ( params->message.temporary_payload_data_length != 1 )
+            {
+                printf( "\r\n\tUnrecognized LED message [IGNORED] Expected 1 or 0" );
+                return;
+            }
+
+            if ( params->message.temporary_payload_data[0] == '0' )
+            {
+                io_led_off();
+            }
+            else
             {
-                topic_conf->on_msg_pull( params->message.temporary_payload_data,
-                                         params->message.temporary_payload_data_length );
+                io_led_on();
             }
             break;
         default:
             return;
     }
 }
 
-void push_mqtt_topics()
+void publish_mqtt_topics()
 {
-    int i = 0;
-    for ( ; i < topics_array_length; ++i )
+    if ( !is_burst_mode )
     {
-        const mqtt_topic_configuration_t* const topic_config = &topics_array[i];
-
-        if ( NULL != topic_config->on_msg_push )
-        {
-            topic_config->on_msg_push( &topic_config->topic_descr );
-        }
+        pub_gyroscope( XI_TOPIC_NAME_MANGLE( "Gyroscope" ) );
+        pub_accelerometer( XI_TOPIC_NAME_MANGLE( "Accelerometer" ) );
+        pub_button( XI_TOPIC_NAME_MANGLE( "Button" ) );

```

Comment:
We need to pub_button() every time so CPM gets a `0` whenever the button stops being pressed, regardless of the state of is_burst_mode.

That is because there's no interrupt configured for the "button released" edge. I don't think it's possible to configure such interrupt for this pin in this device (there's no 'signal edge' parameter in `BSP_PB_Init()`), but I can look into it if it becomes a problem.